.github/workflows/deploy-gh-pages.yml

```diff
@@ -0,0 +1,43 @@
+name: "deploy-github-pages"
+
+on:
+  push:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
```

Comment:
Shall we add
```
        with:
          submodules: "recursive"
```
here or remove it in https://github.com/y-scope/yscope-log-viewer/blob/bc8109f9f371c3327070000dee3537bdaf535428/.github/workflows/lint.yaml#L26-L27

Reply:
I would either add it here or leave both as is. Although we don't need it in lint.yaml currently, hopefully we'll yscope-dev-utils to this repo soon in service of adding more automated linting. So in that case, changing lint.yaml would be unnecessary; I'm not sure whether we would need to change deploy-gh-pages.yml at that point.

Speaking of, I just noticed that we should rename this workflow file with `.yaml`.